common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -106,7 +109,13 @@ private String getStacktrace(Throwable e) {
     }
 
     public void close() {
-
+        try {
+            if(dataPublisher!=null){
+                dataPublisher.shutdown();
+            }
+        } catch (DataEndpointException e) {
+            log.error("Error in shutting down the data publisher " + e.getMessage(), e);
+        }
     }
 
```

Comment:
This method neither return boolean or throwing exception so how to identify whether publisher shutdown correctly or not ?